File: src/main/frontend/src/service/TripService.ts
import {Page} from "../model/Page";
import {Trip} from "../model/Trip";
import {PageParams} from "../model/PageParams";

export interface TripService {
  getTrips(page: PageParams): Promise<Page<Trip>>;

  getTripsByCargo(cargoId: number, page?: number): Promise<Page<Trip>>;

  getTrip(id?: number): Promise<Trip>

  saveTrip(trip: Trip): Promise<Trip>;
}

File: src/main/java/com/raihanorium/springreact/service/TripService.java
package com.raihanorium.springreact.service;

import com.raihanorium.springreact.model.Trip;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.Optional;

public interface TripService {
    Optional<Trip> findById(Long id);

    Page<Trip> findAll(String search, Pageable pageable);

    Page<Trip> findAllByCargoId(Long cargoId, Pageable pageable);

    Trip save(Trip trip);

    void deleteAll();
}
